pkg/secp256k1/signer.go

```diff
@@ -135,18 +135,21 @@ func (k *KeyPair) SignDirect(message []byte) (ethSig *SignatureData, err error)
 	if k == nil {
 		return nil, fmt.Errorf("nil signer")
 	}
-	sig, err := ecdsa.SignCompact(k.PrivateKey, message, false) // uses S256() by default
-	if err == nil {
-		// btcec does all the hard work for us. However, the interface of btcec is such
-		// that we need to unpack the result for Ethereum encoding.
-		ethSig = &SignatureData{
-			V: new(big.Int),
-			R: new(big.Int),
-			S: new(big.Int),
-		}
-		ethSig.V = ethSig.V.SetInt64(int64(sig[0]))
-		ethSig.R = ethSig.R.SetBytes(sig[1:33])
-		ethSig.S = ethSig.S.SetBytes(sig[33:65])
+	sig := ecdsa.SignCompact(k.PrivateKey, message, false) // uses S256() by default
+	if len(sig) < 65 {
```

Comment:
Doesn't  it have to be exactly 65? Why does the new API not throw an error?

Reply:
Is this https://github.com/decred/dcrd/blob/f98d08ef138a99711dbbc86c569935ded8d6a986/dcrec/secp256k1/ecdsa/signature.go#L787-L788 ?

Reply:
https://github.com/btcsuite/btcd/pull/2211/files 

Okay I don't see why we need this check it will always return a signature of length 65